src/runtime_src/core/pcie/common/memaccess.h

```diff
@@ -40,6 +40,17 @@
 #include "xclhal2.h"
 #include "xclbin.h"
 
+class IosBaseFlags {
```

Comment:
lets move this code to runtime_src/core/common/utils.h
Put it under namespace xrt_core { ... }
And calls it `class ios_flags_restore`.